pkg/secrets/repositoryserver/repository_server_useraccess_credentials.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repositoryserver
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ RepositoryServerSecret = &RepositoryServerUserAccessCredentials{}
+
+type RepositoryServerUserAccessCredentials struct {
+	credentials *corev1.Secret
+}
+
+func NewRepositoryServerUserAccessCredentials(secret *corev1.Secret) *RepositoryServerUserAccessCredentials {
+	return &RepositoryServerUserAccessCredentials{
+		credentials: secret,
+	}
+}
+
+func (r *RepositoryServerUserAccessCredentials) Validate() error {
+	return nil
```

Comment:
Do we not need any validation here? Let's add a comment if not.

Reply:
Only validation we need is the length of data. Otherwise the key and values can be anything for the secret